service-api/module/Application/src/Yoti/YotiService.php

```diff
@@ -122,13 +116,194 @@ public function retrieveResults(string $sessionId): array
+                'headers' => $headers,
+                'query' => [
+                    'sdkId' => $this->sdkId->getValue(),
+                    'sessionId' => $yotiSessionId,
+                    'nonce' => $nonce,
+                    'timestamp' => $timestamp
+                ]
+            ]);
+
+            if ($config->getStatusCode() !== Response::STATUS_CODE_200) {
+                $this->logger->error('Case configuration retrieval was unsuccessful ', [
+                    'data' => [ "sessionId" => $yotiSessionId]
+                ]);
+                throw new YotiException("Error: " . $config->getReasonPhrase());
+            }
+            return json_decode(strval($config->getBody()), true);
+        } catch (GuzzleException $e) {
+            $this->logger->error('Unable to connect to Yoti service [' . $e->getMessage() . '] ', [
+                'data' => [ "operation" => "session-configuration", "sessionId" => $yotiSessionId ]
+            ]);
+            throw new YotiException("A connection error occurred. Previous: " . $e->getMessage());
+        }
+    }
+
     /**
      * @param string $sessionId
      * @return array
-     * Generate PDF letter for applicant
+     * PDF Stage 2: Retrieve PDF letter contents
+     * @throws YotiException
      */
-    public function retrieveLetterPDF(string $sessionId): array
+    public function retrieveLetterPDF(string $sessionId, string $nonce, int $timestamp): array
+    {
+        $headers = $this->getSignedRequest(
+            '/sessions/' . $sessionId . '/instructions/pdf',
+            'GET',
+            $nonce,
+            $timestamp,
+            null,
+            $sessionId
+        );
+        try {
+            $pdfData = $this->client->get('/idverify/v1/sessions/' . $sessionId . '/instructions/pdf', [
+                'headers' => $headers,
+                'query' => [
+                    'sdkId' => $this->sdkId->getValue(),
+                    'sessionId' => $sessionId,
+                    'nonce' => $nonce,
+                    'timestamp' => $timestamp
+                ],
+            ]);
+        } catch (GuzzleException $e) {
+            $this->logger->error('Unable to connect to Yoti service [' . $e->getMessage() . '] ', [
+                'data' => [ 'operation' => 'retrieve-pdf', 'header' => $headers]
+            ]);
+            throw new YotiException("A connection error occurred. Previous: " . $e->getMessage());
+        }
+        $base64 = base64_encode(strval($pdfData->getBody()));
+        // Convert base64 to pdf
+        $pdf = base64_decode($base64);
+
+        return ["status" => "PDF Created", "pdfData" => $pdf];
+    }
+
+    public function letterConfigPayload(CaseData $caseData, string $requirementId): array
     {
-        return [];
+        $payload = [];
+        $email = (string) getenv("YOTI_LETTER_EMAIL") ?: "opg-all-team+yoti@digital.justice.gov.uk";
```

Comment:
It would be good to move this into the `YotiServiceFactory`, so that `YotiService` just has a `notificationEmail` constructor parameter rather than using getenv itself directly.